docs/go-sync-vs-async-start.md

```diff
@@ -0,0 +1,35 @@
+---
+id: go-sync-vs-async-start
+title: Sync vs async Workflow start
+sidebar_label: Sync vs async Workflow start
+---
+
+Workflows can be started both synchronously and asynchronously.
+
+## Synchronous start
+
+A Synchronous start initiates a Workflow and then waits for its completion. The started Workflow will not rely on the invocation process and will continue executing even if the waiting process crashes or stops.
+
+In Go, the only difference between a synchronous start and an asynchronous start is whether you use the Workflow Execution entity to get the result of the Workflow from the same process in which you started it.
+
+<!--SNIPSTART hello-world-project-template-go-start-workflow-->
+<!--SNIPEND-->
+
+## Asynchronous start
+
+An asynchronous start initiates a Workflow execution and immediately returns to the caller. This is the most common way to start Workflows in a live environment.
+
+To start a Workflow asynchronously in Go just call `ExecuteWorkflow()` on the client and don't wait for the result in the same process.
+
+<!--SNIPSTART money-transfer-project-template-go-start-workflow-->
+<!--SNIPEND-->
+
+If you need to get the result of a Workflow after an asynchronous start, the most straightforward way is to make a blocking call to the Workflow.
+
+If `WorkflowOptions.WorkflowIdReusePolicy` is not set to `AllowDuplicate`, then instead of throwing `DuplicateWorkflowException`, it reconnects to an existing Workflow and waits for its completion. The following example shows how to do this from a different process than the one that started the Workflow. All this process needs is a `WorkflowId`.
```

Comment:
Maybe it's because I genuinely don't understand this mechanism, but why would workflow id reuse be relevant to getting a running workflow execution. I could understand if the code below executed a wrokflow but it does not.

Also why is this not in snipsync?

Reply:
Checked with Vitaly - and indeed this is how it is done.

Adding a working example is in my back log.